### DISCUSSION TITLE
Frequently Asked Questions

### DISCUSSION THREAD

**Discussion by brichbash** [JetBrains/Qodana#54] [General]
> [!TIP]
> This discussion can be **outdated**, for the latest FAQ see https://www.jetbrains.com/help/qodana/faq.html

### Are there any plans to support `<technology-name>`?

All technologies supported by JetBrains IDEs are going to be eventually covered by Qodana, too. You can create an issue [on our tracker](https://youtrack.jetbrains.com/issues?q=project:%20Qodana%20) or vote for an existing one to let us know what technology we should focus on.

- Scala – [https://youtrack.jetbrains.com/issue/QD-1031](https://youtrack.jetbrains.com/issue/QD-1031)
- Dart – [https://youtrack.jetbrains.com/issue/QD-2226](https://youtrack.jetbrains.com/issue/QD-2226)

### How do I integrate Qodana with `<CI/CD name>`?

Any Qodana linter is a Linux Docker image, so any CI/CD that supports Docker should be able to run it. We are working on extending [our documentation](https://www.jetbrains.com/help/qodana/ci.html) to provide users with the best examples on how to integrate it with different CI/CD platforms. If you experience any difficulties, feel free to contact our support at [qodana-support@jetbrains.com](mailto:qodana-support@jetbrains.com).

### Can I import SARIF of another analysis tool to Qodana?

No. At this time, there is no opportunity to show the results of external analysis tools in Qodana UI, though we're working on it.

### What's a profile?

It's a set of pre-configured checks that include the checks state (enabled/disabled), its options, and the path the checks are applied to. Qodana inspection profiles are the same as [IntelliJ IDEA inspection profiles](https://www.jetbrains.com/help/idea/inspections-settings.html#profile_management) and can be reused.

### Qodana reports 0 errors, though I know it can't be true.

Try using `qodana.recommended` inspection profile. It has more checks than the default one.

If the problem persists, please create an issue in our tracker or contact us at [qodana-support@jetbrains.com](mailto:qodana-support@jetbrains.com) and attach logs (they are stored in `/data/results`, you can access them mounting your directory to the path).

### Is there a way to reduce analysis time?

Yes, you can use [caching](https://www.jetbrains.com/help/qodana/qodana-jvm-community-docker-techs.html#Cache+dependencies) (which is enabled by default in [Qodana Scan GitHub Action](https://github.com/marketplace/actions/qodana-scan)). In case it doesn't help, please create an issue in our tracker or contact us at [qodana-support@jetbrains.com](mailto:qodana-support@jetbrains.com) and attach logs (they are stored in `/data/results`, you can access them mounting your directory to the path or if you are using GitHub actions, they are uploaded to the workflow artifacts).

### How to make Qodana check formatting rules for Kotlin?

All you need to do is to add `IncorrectFormatting` inspection to [qodana.yaml](https://www.jetbrains.com/help/qodana/qodana-yaml.html).

### Does Qodana support custom checks?

At the moment, you can add to the inspection profile your [structural search inspections](https://www.jetbrains.com/help/idea/creating-custom-inspections.html), or you can use plugins that add inspections to IntelliJ-based IDEs. Or you can write your plugin with code inspections ([https://plugins.jetbrains.com/docs/intellij/github-template.html](https://plugins.jetbrains.com/docs/intellij/github-template.html)). We are going to publish a Qodana-related guide very soon.

### Qodana fails with OOM (Out of Memory) error.

See [https://github.com/JetBrains/Qodana/discussions/51](https://github.com/JetBrains/Qodana/discussions/51)

### Qodana can't download Gradle because I use proxy.

Please run `./gradlew` in the root folder before starting Qodana. The downloaded Gradle will be used by Qodana. 
**NB:** If your project was created on Windows, be sure to run `git update-index --chmod=+x gradlew` to make the file executable in your CI.

**Comment by KerryBeetge** [comment]
Code coverage and C++ are now supported.
